engines/osxnative/user.go

```diff
@@ -0,0 +1,149 @@
+package osxnative
+
+import (
+	"github.com/satori/go.uuid"
+	"os"
+	"path"
+	"strconv"
+)
+
+type user struct {
+	d    dscl
+	name string
+}
+
+func (u *user) create() error {
+	uid, err := u.getMaxUid()
+	if err != nil {
+		return err
+	}
+
+	staffGid, err := u.d.read("/Groups/staff", "PrimaryGroupID")
```

Comment:
so really this user could still have the same permissions as the cltbld user, but just running under a different user?  This is a step in the right direction but I think we need to spend time understand what this user in that group is capable of.  We need to makes sure that it can't see things it's not supposed to, change ACLs, or anything else destructive.